File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:

        prefixSum=0
        count=0
        freqCount=defaultdict(int)
        freqCount[0]=1

        for num in nums:
            prefixSum+=num
            count+=freqCount[prefixSum-k]
            freqCount[prefixSum]+=1

        print(freqCount)
        return count
            
    



        

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
      int prefixSum=0;
      Map<Integer,Integer> map=new HashMap<>();
      map.put(0,1);
      int count=0;
      for(int num:nums){
            prefixSum+=num;
            int rem=prefixSum-k;
            count+=map.getOrDefault(rem,0);
            map.put(prefixSum,map.getOrDefault(prefixSum,0)+1);
      }  
      return count;
    }
}